functions/fish_prompt.fish

```diff
@@ -1255,6 +1321,8 @@ function fish_prompt -d 'bobthefish, a fish theme optimized for awesome'
         __bobthefish_prompt_git $git_root_dir $real_pwd
     else if [ "$hg_root_dir" ]
         __bobthefish_prompt_hg $hg_root_dir $real_pwd
+    else if [ "$fossil_root_dir" ]
+        __bobthefish_prompt_fossil $fossil_root_dir $real_pwd
```

Comment:
```fish
switch (path sort -r "$git_root_dir" "$hg_root_dir" "$fossil_root_dir")[1]
  case "$git_root_dir"
    __bobthefish_prompt_git $git_root_dir $real_pwd
  case "$hg_root_dir"
    __bobthefish_prompt_hg $hg_root_dir $real_pwd
  case "$fossil_root_dir"
    __bobthefish_prompt_fossil $fossil_root_dir $real_pwd
end
```

I think this would work, and is probably least messy. Not sure if it's worth it either, but I think this is an elegant enough solution. And it could replace the entire series of ifs/elses, since `path sort` will handle empty paths as if they're above `/`:

```shell
$ path sort -r "git" "" "git/fossil repo"
git/fossil repo
git

$ # previous line intentionally left blank
```